src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1905,22 +1910,170 @@ protected virtual void TagHelperPrefixDirective()
                         errors.Add(duplicateDiagnostic);
                     }
 
-                    return new TagHelperPrefixDirectiveChunkGenerator(prefix, errors);
+                    var parsedDirective = ParseDirective(prefix, Span.Start, TagHelperDirectiveType.TagHelperPrefix, errors);
+
+                    return new TagHelperPrefixDirectiveChunkGenerator(
+                        prefix,
+                        parsedDirective.DirectiveText,
+                        parsedDirective.DirectiveStart,
+                        errors);
                 });
         }
 
+        // Internal for testing.
+        internal void ValidateTagHelperPrefix(
+            string prefix,
+            SourceLocation directiveLocation,
+            List<RazorDiagnostic> diagnostics)
+        {
+            foreach (var character in prefix)
+            {
+                // Prefixes are correlated with tag names, tag names cannot have whitespace.
+                if (char.IsWhiteSpace(character) || InvalidNonWhitespaceNameCharacters.Contains(character))
+                {
+                    diagnostics.Add(
+                        RazorDiagnostic.Create(
+                            new RazorError(
+                                Resources.FormatInvalidTagHelperPrefixValue(SyntaxConstants.CSharp.TagHelperPrefixKeyword, character, prefix),
+                                directiveLocation,
+                                prefix.Length)));
+
+                    return;
+                }
+            }
+        }
+
+        private ParsedDirective ParseDirective(
```

Comment:
Since the majority of this code was copy/paste from the other phase going to look at this minimally. Also it'll be replaced when [this](https://github.com/aspnet/Razor/issues/1635) is completed at a later date.